static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx

```diff
@@ -621,6 +622,148 @@ function InstalledSdkChecklistItem({
   );
 }
 
+function getToolUsedToUploadSourceMaps({
+  releaseUserAgent,
+}: Pick<FrameSourceMapDebuggerData, 'releaseUserAgent'>) {
+  return {
+    sentryCli: releaseUserAgent?.includes('sentry-cli'),
+    vitePlugin: releaseUserAgent?.includes('vite-plugin'),
+    webpackPlugin: releaseUserAgent?.includes('webpack-plugin'),
+    rollupPlugin: releaseUserAgent?.includes('rollup-plugin'),
+    esbuildPlugin: releaseUserAgent?.includes('esbuild-plugin'),
+  };
+}
+
+const pluginConfig: Record<
+  keyof Omit<ReturnType<typeof getToolUsedToUploadSourceMaps>, 'sentryCli'>,
+  {
+    configFile: string;
+    link: string;
+    name: string;
+  }
+> = {
+  vitePlugin: {
+    name: 'Vite',
+    link: 'https://docs.sentry.io/platforms/javascript/sourcemaps/uploading/vite/',
+    configFile: 'vite.config.(js|ts)',
+  },
+  webpackPlugin: {
+    name: 'Webpack',
+    link: 'https://docs.sentry.io/platforms/javascript/sourcemaps/uploading/webpack/',
+    configFile: 'webpack.config.(js|ts)',
+  },
+  rollupPlugin: {
+    name: 'Rollup',
+    link: 'https://docs.sentry.io/platforms/javascript/sourcemaps/uploading/rollup/',
+    configFile: 'rollup.config.(js|ts)',
+  },
+  esbuildPlugin: {
+    name: 'Esbuild',
+    link: 'https://docs.sentry.io/platforms/javascript/sourcemaps/uploading/esbuild/',
+    configFile: 'esbuild.config.(js|ts)',
+  },
+};
+
+function SentryPluginMessage({
+  toolUsedToUploadSourceMaps,
+  sourceMapsDocLinks,
+}: {
+  sourceMapsDocLinks: ReturnType<typeof getSourceMapsDocLinks>;
+  toolUsedToUploadSourceMaps: ReturnType<typeof getToolUsedToUploadSourceMaps>;
+}) {
+  const activePlugin = Object.keys(pluginConfig).find(
+    key =>
+      toolUsedToUploadSourceMaps[
+        key as keyof ReturnType<typeof getToolUsedToUploadSourceMaps>
```

Comment:
feels like you should pull this into a named type somewhere if its going to get used this many times